src/WebJobs.Extensions.DurableTask/Auth/AzureStorageAccountProvider.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using DurableTask.AzureStorage;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask.Options;
+using Microsoft.Extensions.Configuration;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Auth
+{
+    internal sealed class AzureStorageAccountProvider : SimpleStorageAccountProvider
+    {
+        private readonly IStorageCredentialsFactory credentialFactory;
+        private readonly IConfiguration configuration;
+
+        public AzureStorageAccountProvider(
+            IConnectionStringResolver connectionStringResolver,
+            IStorageCredentialsFactory credentialFactory,
+            IConfiguration configuration)
+            : base(connectionStringResolver)
+        {
+            this.credentialFactory = credentialFactory ?? throw new ArgumentNullException(nameof(credentialFactory));
+            this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        }
+
+        public override StorageAccountDetails GetStorageAccountDetails(string name)
+        {
+            if (this.TryResolveConnectionString(name, out string connectionString))
+            {
+                return new StorageAccountDetails { ConnectionString = connectionString };
+            }
+
+            AzureStorageAccountOptions options = this.GetAzureStorageAccountOptions(this.configuration, name);
+            StorageCredentials storageCredentials = this.credentialFactory.CreateAsync(options).EnsureCompleted();
```

Comment:
It is less than ideal to rely on asynchronous APIs in a synchronous context, but unfortunately we need an initial token to pass to the `TokenCredential` (otherwise it may be unresolved by the time we need it!)